### PR TITLE
Added ability to change tray icon by user.

### PR DESCRIPTION
**Pull request by martinrotter**
I added ability to change tray icon by user.

1. Change is propagated to the settings and directly to tray icon.
2. User can change icon via settings dialog.

### CODE REVIEW

**Comment by martinrotter** [comment]
Also, this kind-of solves some issues, for example #39.

File: src/qlipperpreferences.cpp [hunk 1]
```diff
@@ -19,6 +19,9 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #include "qlipperpreferences.h"
 
+QString DEFAULT_ICON_PATH = ":/icons/qlipper.png";
```
Comment (palinek) [inline src/qlipperpreferences.cpp]: - why isn't it `const`?
- how about using `QStringLiteral`?
- when it's needed elsewhere, why not put it into as a static member of `QlipperPreferences`?

File: src/qlipperpreferencesdialog.cpp [hunk 2]
```diff
@@ -73,6 +76,22 @@ QlipperPreferencesDialog::QlipperPreferencesDialog(QWidget *parent) :
...
+    QAction *actUseDefault = new QAction(QIcon(DEFAULT_ICON_PATH), tr("Use default icon"), this);
+
+    iconMenu->addAction(actLoadFromFile);
+    iconMenu->addAction(actUseDefault);
+
+    temporarilyRembemberNewTrayIcon(s->getPathToIcon());
+
+    buttonIconImage->setMenu(iconMenu);
+
+    connect(actLoadFromFile, SIGNAL(triggered()), this, SLOT(selectIconFromFile()));
```
Comment (palinek) [inline src/qlipperpreferencesdialog.cpp]: Please, use the Qt5 notation for signal/slot connection.

File: src/qlipperpreferencesdialog.ui [hunk 3]
```diff
@@ -35,44 +35,13 @@
...
         <width>340</width>
-        <height>217</height>
+        <height>302</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Preferences</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
-       <item row="4" column="0" colspan="2">
```
Comment (palinek) [inline src/qlipperpreferencesdialog.ui]: If you wish to reorder/normalize, please, do it as separate commit to not polute this one (which handles something other).

File: src/qlipperpreferencesdialog.cpp (hunk 2 above)
Comment (martinrotter) [inline src/qlipperpreferencesdialog.cpp]: Rest of the project does not seem to use Qt5 notation, I used what I saw in rest of sources to respect overall code environment.

Comment (palinek) [inline src/qlipperpreferencesdialog.cpp]: Right, but I believe the Qt5 notation should be used (performance etc.) and the old usage changed in all other places in the future.

Comment (martinrotter) [inline src/qlipperpreferencesdialog.cpp]: Right, I may believe it too. Should it be done in separate commits outside this PR? One notation is better than two notations, imho.

**Comment by palinek** [comment]
> I'm not an UI designer, but for me the icon selection button is distracting in the preferences dialog.

> And according to choose the icon, whouldn't it be feasible to let the user choose also some symbolic name to get the icon from current icon theme?

I had another look and I would propose:
- label, edit, icon button (the same size as the line edit as with the `Keyboard Shortcut`)
The user will be able to manualy set the (symbolic) icon in the QLineEdit and also be able to choose a file by the button

File: src/qlipperpreferencesdialog.cpp (hunk 2 above)
Comment (palinek) [inline src/qlipperpreferencesdialog.cpp]: > Should it be done in separate commits outside this PR?

Just change your new `connect`(s) here and it would be great, if you could make PR for all other usages.

**Comment by martinrotter** [comment]
I don't feel that supporting `QIcon::fromTheme(...)` is useful for one single tray icon. What would we do if user changes icon them in his DE? We would have to detect the change to make it look nice. I do not want to write that detection. Someone else would have to.

Most users have one single icon theme which they stick to all the time. They change icons very rarely and for me personally, selecting this particular icon by file is better than by symbolic name via theme engine.

File: src/qlipperpreferencesdialog.ui (hunk 3 above)
Comment (martinrotter) [inline src/qlipperpreferencesdialog.ui]: If you look thoroughly, you will see that aesthetically, nothing visually changed.

File: src/qlipperpreferencesdialog.cpp (hunk 2 above)
Comment (martinrotter) [inline src/qlipperpreferencesdialog.cpp]: Done.

File: src/qlipperpreferencesdialog.ui (hunk 3 above)
Comment (palinek) [inline src/qlipperpreferencesdialog.ui]: > If you look thoroughly, you will see that aesthetically, nothing visually changed.

I know that nothing changed in UI, but there are more changes in the *.ui than what was made (added) to the UI.

Comment (palinek) [inline src/qlipperpreferencesdialog.ui]: > UI file was missing item with horizontal index 2 in your layout. I fixed it.

..and that should go into separate commit

Comment (martinrotter) [inline src/qlipperpreferencesdialog.ui]: Jesus. This was my first PR and literally last one.

Comment (palinek) [inline src/qlipperpreferencesdialog.ui]: > This was my first PR and literally last one.

What's the problem here? You don't seem like a git novice. If this is a realy annoying task for you, I (we) can handle it in behalf of you. But you should say it.... and not just close the PR in anger.

**Comment by martinrotter** [comment]
Git operation is not a problem. I just will not debate over few lines which do not do anything, which do not actually change anything. Feel free to edit PR by yourself. :)
